### DISCUSSION TITLE
Diverging stacked Likert-style bar charts with labels

### DISCUSSION THREAD

**Discussion by spandl** [observablehq/plot#2149] [Q&A] [answered]
There is this really nice example here that creates bar charts for survey results.
https://observablehq.com/@observablehq/plot-diverging-stacked-bar

I was trying to get data labels into the bars, but so far failed.

I was actually able to even remove the z grouping. It still returns the same result.
```
Plot.barX(dataset, {
        y: data.accessors.y,
        x: data.accessors.x,
        
        fill: "rating",
        ...likertScale,
         inset: 1,
        sort: {
            y: "y"
        }
      }),
```

The hick is that `Plot.text` doesn't contain a property `offset`, like bars.

Is there an easy way to position the labels, based on another mark or are marks completely separate?
Or is the only way to achieve this to use `render()` and calculate the positions within a channel?

Data passed to `offset: [FUNCTION]` or `channel: { channelName: [Function]}` aren't the same. But this would eventually be a way to use one function to calculate offsets for bars and labels.

## Challenge: place labels on stacked bars that have been moved with an offset function
![image](https://github.com/user-attachments/assets/8372c18e-4d3b-4f7b-aa54-c9545e3074fd)

**Comment by Fil** [comment]
Note that the last cell at the bottom of https://observablehq.com/@observablehq/plot-diverging-stacked-bar has text labels; if you replace `text: "first"` by `text: "count"` it should display the values instead?

**Comment by spandl** [comment, marked as answer]
Oh.

All I needed to do was adding:

`z: "rating",`

So my text mark would be able to process the Likert function.
My example was a tick different, as I don't use groupZ.

Well, that was a good exercise in deep-diving Plot!

Thanks @Fil, awesome support here !!

![image](https://github.com/user-attachments/assets/a029d9bc-a5d0-4f38-b031-6ac758fb885a)
